--- a/alsa/stdioemu.c
+++ b/alsa/stdioemu.c
@@ -37,7 +37,9 @@
 #endif
 
 #include <stdio.h>
+#ifdef HAVE_LIBIO_H
 #include <libio.h>
+#endif
 
 struct fd_cookie {
 	int fd;
@@ -99,7 +101,11 @@ static FILE *fake_fopen(const char *path
 
 	if (open_mode && fdc->fd > 0) {
 		result = fopencookie (fdc,"w", fns);
+#ifdef HAVE_FILENO
 		result->_fileno = fdc->fd;		/* ugly patchy slimy kludgy hack */
+#else
+		result->__filedes = fdc->fd;
+#endif
 	}
 	return result;
 }
